packages/common/src/Adaptor.js

```diff
@@ -932,3 +932,53 @@ export function assert(expression, errorMessage) {
     return state;
   };
 }
+
+/**
+ * Outputs a message to the console while maintaining the state.
+ * @public
+ * @function
+ * @example <caption>Log specific values</caption>
+ * log('Hello', 'World')
+ * // Logs: Hello World
+ * log($.user.name)
+ * // Logs value of state.user.name
+ * @example <caption>Log a specific object from state</caption>
+ * log(({ configuration, ...rest }) => rest);
+ * @example <caption>Multiple arguments including state references</caption>
+ * log('User:', $.user.name, 'Age:', $.user.age)
+ * // Logs: User: John Age: 30
+ * @param {...*} args - Values to log: primitives, objects, or state references
```

Comment:
No-one knows what a state reference is.

tbh I'd lie a little bit and do this:
```
@param {any} value - A value or message to display in the logs
```

It's true that you can pass multiple things (like in console.log) - but 90% of use cases only require a single argument. And by documenting this use-case, we make life easier for everyone

Don't change the implementation though